boards/qorvo/decawave_dwm3001cdk/decawave_dwm3001cdk.dts

```diff
@@ -0,0 +1,170 @@
+ */
+
+/dts-v1/;
+#include <nordic/nrf52833_qdaa.dtsi>
+#include "decawave_dwm3001cdk-pinctrl.dtsi"
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+/ {
+	model = "Decawave DWM3001CDK";
+	compatible = "decawave,dwm3001";
+
+	chosen {
+		zephyr,console        = &cdc_acm_uart0;
+		zephyr,shell-uart     = &cdc_acm_uart0;
+		zephyr,uart-mcumgr    = &cdc_acm_uart0;
+		zephyr,bt-mon-uart    = &cdc_acm_uart0;
+		zephyr,bt-c2h-uart    = &cdc_acm_uart0;
+		zephyr,sram           = &sram0;
+		zephyr,flash          = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,ieee802154     = &ieee802154;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		// led from top of board down
+		// D20: soemthing related to jlink, red
+		// D13: DW3000 tx(red)/rx(green)
+		// green
+		led0: led_0 {
+			gpios = <&gpio0 04 GPIO_ACTIVE_LOW>;
+			label = "D9 LED";
+		};
+		// red
+		led1: led_1 {
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+			label = "D12 LED";
+		};
+		// red
+		led2: led_2 {
+			gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+			label = "D11 LED";
+		};
+		// blue
+		led3: led_3 {
+			gpios = <&gpio0 05 GPIO_ACTIVE_LOW>;
+			label = "D10 LED";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		// SW1 is connected to P0.18, whicy by default is nRESET and
+		// will reset the board
+		button1: button_1 {
+			gpios = <&gpio0 18 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button switch 1";
+			zephyr,code = <INPUT_KEY_0>;
+		};
+		button2: button_2 {
+			gpios = <&gpio0 2 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button switch 2";
+			zephyr,code = <INPUT_KEY_1>;
+		};
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		sw0        = &button1;
+		sw1        = &button2;
+		led0       = &led0;
+		led1       = &led1;
+		led2       = &led2;
+		led3       = &led3;
+		watchdog0  = &wdt0;
+		accel0     = &lis2dh12;
+	};
+
+};
+
+&uicr {
+	gpio-as-nreset;
+};
+
+&gpiote {
+	status = "okay";
+};
+
+&gpio0 {
+	status = "okay";
+};
+&gpio1 {
+	status = "okay";
+};
+
+&i2c0 {
+	compatible = "nordic,nrf-twim";
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_FAST>;
+
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-1 = <&i2c0_sleep>;
+	pinctrl-names = "default", "sleep";
+	lis2dh12: lis2dh12@19 {
+		compatible = "st,lis2dh12", "st,lis2dh";
+		reg = <0x19>;
+		irq-gpios = <&gpio0 16 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&spi3 {
+	compatible = "nordic,nrf-spim";
+	status = "okay";
+	cs-gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
+
+	pinctrl-0 = <&spi3_default>;
+	pinctrl-1 = <&spi3_sleep>;
+	pinctrl-names = "default", "sleep";
+};
+
+&ieee802154 {
+	status = "okay";
+};
+
+// flash from nrf52833dk_nrf52833
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 0xC000>;
+		};
+		slot0_partition: partition@c000 {
+			label = "image-0";
+			reg = <0x0000C000 0x37000>;
```

Comment:
```suggestion
			reg = <0x0000C000 0x38000>;
```

Reply:
I took these from the nrf52833dk_nrf52833 config, why should it be different here?

Reply:
The ones you based it on are non-optimal, they'll work, sure, but if you use the swap using move algorithm then you will use 1 sector in the primary slot to move the sectors up by, and another for the swap status, that means your maximum image size is reduced by 2 sectors (2*4096 bytes = 8192, 0x2000). So by shifting one sector from the secondary to the primary here, you increase the maximum size of an application from being 0x35000 to 0x36000

Reply:
```
		boot_partition: partition@0 {
			label = "mcuboot";
			reg = <0x00000000 0xC000>;
		};
		slot0_partition: partition@c000 {
			label = "image-0";
			reg = <0x0000C000 0x38000>;
		};
		slot1_partition: partition@44000 {
			label = "image-1";
			reg = <0x00044000 0x36000>;
		};
		storage_partition: partition@7a000 {
			label = "storage";
			reg = <0x0007A000 0x00006000>;
		};
```